service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -238,40 +241,12 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
   @Override
   public ResponseEntity<AbortRunSetResponse> abortRunSet(UUID runSetId) {
     AbortRunSetResponse aborted = new AbortRunSetResponse();
-    List<String> failedRunIds = new ArrayList<>();
-
-    // Get the run set associated with runSetId
-    RunSet runSet = runSetDao.getRunSet(runSetId);
-    // Update the run set to have a canceling state
-    runSetDao.updateStateAndRunDetails(
-        runSetId,
-        CbasRunSetStatus.CANCELING,
-        runSet.runCount(),
-        runSet.errorCount(),
-        OffsetDateTime.now());
 
     aborted.runSetId(runSetId);
 
-    // Get a list of workflows able to be canceled
-    List<Run> runningWorkflows =
-        runDao.getRuns(new RunDao.RunsFilters(runSetId, NON_TERMINAL_STATES));
-    List<UUID> submittedAbortWorkflows = new ArrayList<>();
-
-    for (Run run : runningWorkflows) {
-      // Trying inside the for-loop in case a single run fails to be updated
-      try {
-        cromwellService.cancelRun(run);
-        submittedAbortWorkflows.add(run.runId());
-        // Update each run to have a 'canceling' run state
-        runDao.updateRunStatus(run.runId(), CbasRunStatus.CANCELING, OffsetDateTime.now());
-      } catch (cromwell.client.ApiException e) {
-        String msg = "Unable to abort workflow %s.".formatted(run.runId());
-        log.error(msg, e);
-        failedRunIds.add(run.runId().toString());
-        // Add the error message against the run
-        run.withErrorMessage(msg);
-      }
-    }
+    Map<String, List<String>> abortDetails = abortManager.abortRunSet(runSetId);
```

Comment:
It might be a good idea to convert abort details `Map<String, List<String>>` into its own class called `AbortRequestDetails` with 2 properties `abortRequestFailedIds` and `abortRequestSubmittedIds`. This way it would be easier to access their values rather than using the Map key.